rancher-sandbox/rancher-desktop-agent
# Add docker event monitor

This PR adds a mechanism to watch Docker's stream API for container start/stop events. Also, it stores the port mappings to send over to the host.

To test: 

Build
```
GOOS=linux CGO_ENABLED=0 go build -ldflags=-w ./cmd/rancher-desktop-guestagent/
```
Run on a wsl distro

```
./rancher-desktop-guestagent -debug=true -docker=true
```

Run and verify debug logs 

```
run -d --name test-nginx2 -p 38082:80 nginx
```
```
docker stop [containerID]
```
